services/etcd.go

```diff
@@ -449,18 +450,30 @@ func RunEtcdSnapshotSave(ctx context.Context, etcdHost *hosts.Host, prsMap map[s
 		if err := docker.DoRemoveContainer(ctx, etcdHost.DClient, EtcdSnapshotOnceContainerName, etcdHost.Address); err != nil {
 			return err
 		}
+
+		// If the etcd container is not running the snapshot will never succeed
+		log.Debugf(ctx, "[etcd] Checking if etcd is running on host [%s]", etcdHost.Address)
+		if running, err := docker.IsContainerRunning(ctx, etcdHost.DClient, etcdHost.Address, "etcd", true); err != nil {
```

Comment:
Update to

```
// If the etcd container does not exist the snapshot will never succeed
if exist, err := docker.doesContainerExist...
```

Reply:
This is actually intended. The etcd container has to be running or the snapshot will fail (that was actually the original code I had, that didn't work).

Reply:
This appears to be the only case where updating `IsContainerRunning` to `doesContainerExist` doesn't make sense. Maybe still update the var and func name, and leave the original comment about the container needing to be running? That's my 2c

Reply:
There may be some confusion around this so let me clarify: The original function named `IsContainerRunning` wasn't actually checking if the container was running, rather that it existed. In this case we want to make sure the container itself is running (not stopped, or killed, or starting, etc.) so I renamed the previous function to what it actually does, and introduced a new function `IsContainerRunning` that actually checks that the container does have the state `running`.